src/chainstate/coordinator/mod.rs

```diff
@@ -658,6 +704,287 @@ impl<
         Ok(())
     }
 
+    /// This function reads rejection-related information from the exit-at-rc clarity contract.
+    /// It returns true if the rejection succeeded, and false if it failed.
+    pub fn read_rejection_state(
+        &mut self,
+        rc_cycle_of_rejection: u64,
+        proposed_exit_rc: u64,
+        consensus_hash: &ConsensusHash,
+        block_hash: &BlockHeaderHash,
+        exit_contract_id: &QualifiedContractIdentifier,
+    ) -> Result<bool, Error> {
+        let stacks_block_id = StacksBlockHeader::make_index_block_hash(consensus_hash, block_hash);
+        let reward_cycle_length = self.burnchain.pox_constants.reward_cycle_length;
+        let rejection_percent_threshold = self
+            .burnchain
+            .exit_contract_constants
+            .rejection_confirmation_percent;
+        let is_mainnet = self.burnchain.is_mainnet();
```

Comment:
It doesn't look like `is_mainnet` is ever used in this method -- can be removed.